tests/cstor/snapshot/snapshot_test.go

```diff
@@ -0,0 +1,187 @@
+
+			By("building spc object")
+			spcObj = spc.NewBuilder().
+				WithName(spcName).
+				WithDiskType(string(apis.TypeSparseCPV)).
+				WithMaxPool(1).
+				WithOverProvisioning(false).
+				WithPoolType(string(apis.PoolTypeStripedCPV)).
+				Build().Object
+
+			By("creating storagepoolclaim")
+			_, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc", spcName)
+
+			By("verifying healthy csp count")
+			cspCount := ops.GetHealthyCSPCount(spcName, 1)
+			Expect(cspCount).To(Equal(1), "while checking cstorpool health count")
+
+		})
+	})
+
+	AfterEach(func() {
+		By("deleting resources created for cstor volume snapshot provisioning", func() {
+			By("listing spc")
+			spcList, err = ops.SPCClient.List(metav1.ListOptions{})
+			Expect(err).To(BeNil(), "while listing spc clients", spcList)
+
+			By("deleting spc")
+			for _, spc := range spcList.Items {
+				_, err = ops.SPCClient.Delete(spc.Name, &metav1.DeleteOptions{})
+				Expect(err).To(BeNil(), "while deleting the spc's", spc)
+			}
+			By("deleting storageclass")
+			err = ops.SCClient.Delete(scName, &metav1.DeleteOptions{})
+			Expect(err).To(BeNil(), "while deleting storageclass", scName)
+
+		})
+	})
+
+	When("cstor pvc with replicacount 1 is created", func() {
+		It("should create cstor volume target pod", func() {
+
+			By("building a pvc")
+			pvcObj, err = pvc.NewBuilder().
+				WithName(pvcName).
+				WithNamespace(nsName).
+				WithStorageClass(scName).
+				WithAccessModes(accessModes).
+				WithCapacity(capacity).Build()
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building pvc {%s} in namespace {%s}",
+				pvcName,
+				nsName,
+			)
+
+			By("creating above pvc")
+			_, err = ops.PVCClient.WithNamespace(nsName).Create(pvcObj)
+			Expect(err).To(
+				BeNil(),
+				"while creating pvc {%s} in namespace {%s}",
+				pvcName,
+				nsName,
+			)
+
+			By("verifying target pod count as 1")
+			controllerPodCount := ops.GetPodRunningCountEventually(openebsNamespace, targetLabel, 1)
+			Expect(controllerPodCount).To(Equal(1), "while checking controller pod count")
+
+			By("verifying pvc status as bound")
+			status := ops.IsPVCBoundEventually(pvcName)
+			Expect(status).To(Equal(true), "while checking status equal to bound")
+
+			By("building a cstor volume snapshot")
+			snapObj, err = snap.NewBuilder().
+				WithName(snapName).
+				WithNamespace(nsName).
+				WithPVC(pvcName).
+				Build()
+			Expect(err).To(
+				BeNil(),
+				"while building snapshot {%s} in namespace {%s}",
+				snapName,
+				nsName,
+			)
+
+			By("creating cstor volume snapshot")
+			_, err = ops.SnapClient.WithNamespace(nsName).Create(snapObj)
+			Expect(err).To(
+				BeNil(),
+				"while creating snapshot {%s} in namespace {%s}",
+				snapName,
+				nsName,
+			)
+
+			By("verifying type as ready")
+			snaptype := ops.GetSnapshotTypeEventually(snapName)
+			Expect(snaptype).To(Equal("Ready"), "while checking snapshot type")
```

Comment:
In the next PR we need to think about getting the snapshot details from cstor pool.
In other words, we need to start thinking about using storage engine's APIs in addition to Kubernetes APIs